doc/design.md

```diff
@@ -238,7 +238,46 @@ Returns: New user
     * user_id
     * email
     * name
-    * group
+    * group_name
+    * admin
+    * timestamp
+
+
+Status codes:
+
+- `201 CREATED`: user creation was successful
+- `400 BAD FORMAT`: missing one or more fields
+- `401 UNAUTHORIZED`: authorization rejected; missing or invalid auth token
+- `403 FORBIDDEN`: authorized user is not admin
+- `422 UPROCESSABLE ENTITY`: duplicate user or password too short
+
+
+**`PUT`** `/api/users/<user_id>`
+
+Updates one or more data fields for an existing user. Requires admin authorization or that the logged-in user's id match the requested user_id.
+
+Required headers:
+
+- `Authorization: <JWT string>`
+
+`PUT` body:
+
+- format: `JSON`
+- fields (one or more of):
+    * email
+    * name
+    * group_name
+    * password
+    * admin
+
+Returns: Updated user
+
+- format: `JSON`
+- fields:
+    * user_id
+    * email
+    * name
+    * group_name
     * admin
     * timestamp
 
```

Comment:
What status codes does PUT return?

Reply:
Ah, yes I've updated the status codes for PUT.